### ISSUE TITLE
Wrong async generation after a null coalesce in 0.8.2.11

### ISSUE THREAD

**Issue by fredericDelaporte** [maca88/AsyncGenerator#133]
Following code:
```c#
cacheBatcher?.ExecuteBatch();
PutCacheableResults();
```

Is generated with 0.8.2.11 as:
```
if (executeBatchTask != null)
{
	await (executeBatchTask).ConfigureAwait(false);
	await (PutCacheableResultsAsync(cancellationToken)).ConfigureAwait(false);
}
```
The null check has taken then next call, which it should not do.

This has been seen by running the async generator on the 5.2.x branch. It does not seem to happen with 0.14.0 or 0.1.5.0 or 0.16.2, but I have not used the exact same code to check that, and I have not seen an issue matching the trouble I report here.
Oddly, the trouble also occurs with 0.8.2.9, although the PR having introduced it was using that version.

**Comment by maca88** [comment]
I've backported the new logic from #117, which fixes the issue and released `0.8.2.12`.

**Comment by fredericDelaporte** [comment]
Thanks!
